### PR TITLE
[code-coverage] Add tests for history decision task handler

### PR DESCRIPTION
**Pull request by ketsiambaku**
**What changed?**
Introduced the following tests in service/history/decision:

- TestHandleDecisionContinueAsNewWorkflow
- TestHandleDecisions
- TestValidateAttributes

**Why?**
improve code coverage in the package

**How did you test it?**

**Potential risks**

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: service/history/decision/task_handler_test.go [hunk 1]
```diff
@@ -1131,6 +1134,200 @@ func TestHandleDecisionFailWorkflow(t *testing.T) {
...
+		taskHandler.sizeLimitChecker.historyCountLimitError = 10
+		taskHandler.mutableState.(*execution.MockMutableState).EXPECT().GetNextEventID().Return(int64(12)) //nextEventID - 1 > historyCountLimit of 10
+		taskHandler.mutableState.(*execution.MockMutableState).EXPECT().GetExecutionInfo().Return(&persistence.WorkflowExecutionInfo{})
+		taskHandler.mutableState.(*execution.MockMutableState).EXPECT().AddFailWorkflowEvent(taskHandler.sizeLimitChecker.completedID, &types.FailWorkflowExecutionDecisionAttributes{
+			Reason:  common.StringPtr(common.FailureReasonSizeExceedsLimit),
+			Details: []byte("Workflow history size / count exceeds limit."),
+		}).Return(nil, errors.New("some error adding fail workflow event"))
+		res, err := taskHandler.handleDecisions(context.Background(), []byte{}, []*types.Decision{})
+		assert.Nil(t, res)
+		assert.NotNil(t, err)
```
Comment (dkrotx) [inline service/history/decision/task_handler_test.go:1326]: nit: it's better to always check err first. I think, functions like `assert.ErrorContains()` will do both.

File: service/history/decision/task_handler_test.go [hunk 2]
```diff
@@ -1131,6 +1134,200 @@ func TestHandleDecisionFailWorkflow(t *testing.T) {
...
+				taskHandler.mutableState.(*execution.MockMutableState).EXPECT().GetExecutionInfo().Return(executionInfo).Times(2)
+				taskHandler.attrValidator.domainCache.(*cache.MockDomainCache).EXPECT().GetDomainName(testdata.DomainID).Return(testdata.DomainName, nil)
+				taskHandler.mutableState.(*execution.MockMutableState).EXPECT().AddFailWorkflowEvent(testTaskCompletedID, &types.FailWorkflowExecutionDecisionAttributes{
+					Reason:  common.StringPtr(common.FailureReasonDecisionBlobSizeExceedsLimit),
+					Details: []byte("ContinueAsNewWorkflowExecutionDecisionAttributes. Input exceeds size limit."),
+				})
+			},
+			asserts: func(t *testing.T, taskHandler *taskHandlerImpl, attr *types.ContinueAsNewWorkflowExecutionDecisionAttributes, err error) {
+				assert.True(t, taskHandler.stopProcessing)
+				assert.Nil(t, err)
```
Comment (dkrotx) [inline service/history/decision/task_handler_test.go:1196]: prefer `assert.NoError()` instead

**Review by davidporter-id-au — APPROVED** [review]

File: service/history/decision/task_handler_test.go (hunk 1 above)
Comment (ketsiambaku) [reply service/history/decision/task_handler_test.go:1326]: Good to know 👍🏾

File: service/history/decision/task_handler_test.go [hunk 3]
```diff
@@ -1131,6 +1134,200 @@ func TestHandleDecisionFailWorkflow(t *testing.T) {
...
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			taskHandler := newTaskHandlerForTest(t)
+			if test.expectMockCalls != nil {
+				test.expectMockCalls(taskHandler, test.attributes)
+			}
+			decision := &types.Decision{
+				DecisionType: func(i int32) *types.DecisionType {
```
[the remark this answers is not in the document]
Comment (ketsiambaku) [reply service/history/decision/task_handler_test.go:1290]: Noice. thank you! changing for the other tests as well 👍🏾

**Pull request merged**